loader_hub/google_drive/base.py

```diff
@@ -176,19 +202,42 @@ def _download_file(self, fileid: str, filename: str) -> str:
         Returns:
             The downloaded filename, which which may have a new extension
         """
+
+        from io import BytesIO
+        from googleapiclient.http import MediaIoBaseDownload
+        from googleapiclient.discovery import build
+
         try:
-            file = self._drive.CreateFile({"id": fileid})
-            if file["mimeType"] in self._mimetypes:
-                download_mimetype = self._mimetypes[file["mimeType"]]["mimetype"]
-                download_extension = self._mimetypes[file["mimeType"]]["extension"]
-                new_filename = filename + download_extension
-                # download file with filename and mimetype
-                file.GetContentFile(new_filename, mimetype=download_mimetype)
-                return new_filename
+            # Get file details
+            service = build("drive", "v3", credentials=self._creds)
+            file = service.files().get(fileId=fileid, supportsAllDrives=True).execute()
```

Comment:
out of curiosity, why do this instead of self._drive.CreateFile?

Reply:
Because [pydrive](https://github.com/googlearchive/PyDrive) has been deprecated and doesn't support Shared Drives. We might wanna replace pydrive entirely

Reply:
makes sense